src/benchmarks/micro/runtime/Benchstones/BenchF/Regula.cs

```diff
@@ -15,13 +15,8 @@ public class Regula
 {
     public const int Iterations = 4000000;
 
-    public static volatile object VolatileObject;
-
     [MethodImpl(MethodImplOptions.NoInlining)]
-    private static void Escape(object obj)
-    {
-        VolatileObject = obj;
-    }
+    private static void Escape(object _) { }
```

Comment:
should we have a common `Escape()` method that can be used everywhere?

Reply:
Also, just to confirm, have you verified the disasm to make sure this does what we intent to do?

Reply:
> Also, just to confirm, have you verified the disasm to make sure this does what we intent to do?

Passing an object to a method that can't be inlined is an old trick used even by [GC.KeepAlive](https://github.com/dotnet/runtime/blob/71a216819169a7ce609326e9404eacf8fb267637/src/coreclr/src/System.Private.CoreLib/src/System/GC.cs#L273)

Reply:
> should we have a common Escape() method that can be used everywhere?

I would prefer to stick with the current guidelines that basically say that all you need to do to prevent from dead code elimination is to return the result from the benchmark and BDN is going to consume it for you

https://github.com/dotnet/performance/blob/6ce5b33651f6fe4ea27befec585e7d68d1f022db/docs/microbenchmark-design-guidelines.md#dead-code-elimination

when I was porting the benchmarks long time ago I was asked to not rerwrite them, this is the only reason why this old benchmark is still using this old pattern.